handbook/engineering/hiring/software-engineer-code-intelligence-indexing.md

```diff
@@ -0,0 +1,84 @@
+* 3-4 engineers transforming and exposing the LSIF data, optimizing queries to be as fast as possible.
+
+
+
+## Qualifications
+
+* Solid understanding of either C++ or Java with a focus on Clang or Javac fundamentals from a language tooling perspective.
+* Experience with dependency management systems and build systems coupled with a strong desire to learn new ones.
+* Ability to communicate clearly and empathetically, especially in writing and documentation.
+* Practiced at creating high quality software balanced with a pragmatic understanding of how to make appropriate tradeoffs (e.g., cut scope) to ship quickly and iterate when necessary.
+
+## Nice-to-haves
+
+* Experience in the developer productivity space.
+* Experience or familiarity with [LSP](https://microsoft.github.io/language-server-protocol/) or [LSIF](https://lsif.dev/).
+* Experience or familiarity with Bazel/Pants and distributed builds.
+* Experience working on small high-performing teams, preferably tech startups.
+
+## Not sure if this is you?
+
+We want a diverse, global team, with a broad range of experience and perspectives. If this job sounds great, but you’re not sure if you qualify, apply anyway! We carefully consider every application, and will either move forward with you, find another team that might be a better fit, keep in touch for future opportunities, or thank you for your time.
+
+## Learn more about us
+
+To create a product that serves the needs of all developers, we are building a diverse [all-remote team](https://about.sourcegraph.com/company/remote) that is [distributed across the world](https://about.sourcegraph.com/company/team). Sourcegraph is an equal opportunity workplace; we welcome people from all backgrounds and communities.
+
+We provide [competitive compensation](https://about.sourcegraph.com/handbook/people-ops/compensation) and [practical benefits](https://about.sourcegraph.com/handbook/people-ops/benefits-and-perks) to keep you happy and healthy so that you can do your best work.
+
+Learn more about what it is like to work at Sourcegraph by reading [our handbook](https://about.sourcegraph.com/handbook).
+
+## Interview process
+
+1. The **Hiring Manager** reviews all the information you provided on your application to determine if you meet our qualifications for this role (if there is another open role we think you would be better qualified for, we will let you know).
+2. A **Recruiter** sets up a 30 minute call to learn more about what you are looking for, tell you about Sourcegraph, and answer any questions that you have.
+3. We set up a 1 hour interview with the **Hiring Manager**, who will tell you more about the role in the team. They will ask you about your past work and accomplishments and assess your suitability for the role.
+4. We schedule the following additional interviews, in no particular order, across multiple days: 
+   - TODO, (coding challenge or walk-through of a project?)
```

Comment:
@efritz @Strum355 For this role focusing on the indexer side, what project do we want to do to assess technical ability?

Reply:
That's difficult since we can't be sure all good candidates would have a huge codeintel-related OSS project to walk through :(

Also _writing_ a code challenge from scratch is going to take a very non-trivial amount of effort. I'm unsure of what our path forward here is.

Reply:
What about if they don't have a related OSS project we ask them to do a walk through of a codeintel related/compiler/tooling project in either C++ or Java?

Reply:
How will they explain trade-offs? Can we be certain that they _know_ deeply about one of these things? Is the prompt just "Choose public code and explain it to us?"

Reply:
Yeah I'm not sure of the exact path forward here.  I'll add a line to our sync for a team chat on it.